pageserver/src/tenant/timeline.rs

```diff
@@ -5451,17 +5451,21 @@ impl Timeline {
                 !(a.end <= b.start || b.end <= a.start)
             }
 
-            let guard = self.layers.read().await;
-            for layer in guard.layer_map()?.iter_historic_layers() {
-                if layer.is_delta()
-                    && overlaps_with(&layer.lsn_range, &deltas.lsn_range)
-                    && layer.lsn_range != deltas.lsn_range
-                {
-                    // If a delta layer overlaps with another delta layer AND their LSN range is not the same, panic
-                    panic!(
+            if deltas.key_range.start.next() != deltas.key_range.end {
+                // skip single-key layer files
```

Comment:
why should we skip them? we still want the lsn ranges to not overlap.

Reply:
LSN ranges will overlap if there are single-key layers produced by legacy compaction. Maybe overlap is not the most accurate word here. See the figure below:

![image](https://github.com/user-attachments/assets/4217d881-784b-4bc1-b776-e356e06a98b3)

This is a valid layer map only when the first + third columns are single key layer files.

Reply:
but we had this case before as well, no? Why do we hit the panic now?